### PR TITLE
Delete differential drive

### PR DESCRIPTION
**Pull request by lukicdarkoo**
Delete deprecated differential drive module. I am changing some variables related to the module in #2740, so I guess it is better to delete it now than later.

### CODE REVIEW

File: lib/controller/matlab/WB_NODE_DIFFERENTIAL_WHEELS.m
Comment (stefaniapedrazzi) [inline lib/controller/matlab/WB_NODE_DIFFERENTIAL_WHEELS.m:2]: Note that after deleting the `WB_NODE_DIFFERENTIAL_WHEELS` node, all the subsequent node values changes and the whole Matlab API should be updated and not just this file.
So you should regenerate the Matlab API files and commit all the changed ones.

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Everything seems fine.
Here are just a couple of files that still needs to be updated:
* `robot.md` documentation of `wb_robot_get_type`:
    https://github.com/cyberbotics/webots/blob/72fc39fe3b97ebc91799df83c1138c785e1d98b3/docs/reference/robot.md#wb_robot_get_type
* `supervisor.md` function `wb_supervisor_node_get_type`: in the note for C++, Java and Python we should use a different node than `Node::DIFFERENTIAL_WHEELS` for the example
    https://github.com/cyberbotics/webots/pull/2749/files#diff-bc04135433036246a5befef3ec0c4113f48bb6f62f0042ae6912e3d43751d63bR566
- in `projects/samples/curriculum/controllers/intermediate_lfm/intermediate_lfm.c` and `projects/samples/curriculum/controllers/novice_ir_sensors/novice_ir_sensors.c` some comments still refer to the `wb_differential_wheels_*` functions
- in the generic window CSS: `generic.css` there are some rules for class "differential-wheels-button" that are no longer used
- in `WbSolid::updateJointChildren` there is a comment referring specifically to differential wheels. It would be good to know if this also applied to other cases or not:
    https://github.com/cyberbotics/webots/blob/cleanup-differential-drive/src/webots/nodes/WbSolid.cpp#L1218-L1222
- `resources/webots_ros/src/robot_information_parser.cpp`:
   https://github.com/cyberbotics/webots_ros/blob/master/src/robot_information_parser.cpp#L96

**Comment by lukicdarkoo** [comment]
> in projects/samples/curriculum/controllers/intermediate_lfm/intermediate_lfm.c and projects/samples/curriculum/controllers/novice_ir_sensors/novice_ir_sensors.c some comments still refer to the wb_differential_wheels_* functions

It looks those comments have been outdated for a while :)

**Comment by lukicdarkoo** [comment]
I reverted kinematics mode (I wasn't aware it exists)

**Review by omichel — CHANGES_REQUESTED** [review]
I tested the two examples with kinematic differential drives (e-puck and khepera) and they both work nicely.

**Review by omichel — DISMISSED** [review]
Thank you. It's all good for me.
Please wait for the review of @stefaniapedrazzi before merging.

**Review by stefaniapedrazzi — DISMISSED** [review]
Thank you.

I will approve this PR again just after the conflicts with the `webots_ros` submodule are fixed.

**Comment by lukicdarkoo** [comment]
@stefaniapedrazzi No conflicts, the tests should pass

**Review by omichel — APPROVED** [review]

**Review by stefaniapedrazzi — APPROVED** [review]

**Comment by eze1376** [comment]
Hi,
When I want to build webots, feature-robocup-controllers branch from source.
I get this error
`make[1]: *** No rule to make target 'nodes/WbDifferentialWheels.hpp', needed by 'WbSolid.o'.  Stop.`
and I think it's because of this PR.
If there is a solution, please let me know!
Thanks,

**Comment by eze1376** [comment]
It will be solve by:
`make clean`

**Pull request merged**
